src/libfsm/endids.c

```diff
@@ -880,6 +880,8 @@ fsm_endid_iter_bulk(const struct fsm *fsm,
 		return 1;
 	}
 
+	const fsm_state_t state_count = fsm_countstates(fsm);
```

Comment:
let's have this inside `#ifndef NDEBUG` if it's just here for the assertion

Reply:
Fixed in 8487fcb1.